docs/DeveloperGuide.md

```diff
@@ -1,38 +1,312 @@
+Allows management of recipes, ingredients and weekly meal plan faster than a typical mouse/GUI
+driven app.
 
-## User Stories
+### User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ... | I want to ...                                     | So that I can ...                                                 |
+|---------|----------|---------------------------------------------------|-------------------------------------------------------------------|
+| v1.0    | user     | add my own recipes to the list                    | refer to them when next time                                      |
+| v1.0    | user     | edit the existing recipe                          |                                                                   |
+| v1.0    | user     | delete a recipe from the list                     | clear the unused recipes                                          |
+| v1.0    | user     | view ingredients of the recipe                    | know what is needed to be prepared                                |
+| v1.0    | user     | list all recipes I have                           | know what I have some idea of what to cook                        |
+| v1.0    | user     | find the recipe that contain specific ingredients | find specific recipe without having to go through the entire list |
+| v1.0    | user     | exit from the program                             |                                                                   |
+| v2.0    | new user | list all the command that can be used             | know what command I can use                                       |
+| v2.0    | user     | add meals I plan to make for the week             | refer to the weekly meals plan next time                          |
+| v2.0    | user     | delete meals I plan to make for the week          | remove some meals from the weekly plan if I change my mind        |
+| v2.0    | user     | categorise recipes using tags                     | group recipes with similar theme together                         |
+| v2.0    | user     | list the recipes by tag                           | list recipes that are under the specific category                 |
 
-## Non-Functional Requirements
+### Non-Functional Requirements
 
 {Give non-functional requirements}
 
-## Glossary
+### Glossary
 
 * *glossary item* - Definition
 
-## Instructions for manual testing
+### Instructions for manual testing
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+{Give instructions on how to do a manual product testing e.g., how to load sample data to be used
+for testing}
```

Comment:
Perhaps it might be better to give a more detailed description as to how to perform manual testing, glossary, or remove this section completely.